File: cpp/08_032.cpp
#include <vector>
#include <stack>

std::vector<int> daysUntilWarmer(
    std::vector<int>& temperatures) {
    
    int n = temperatures.size();
    std::vector<int> result(n, 0);
    std::stack<int> stack;
    
    for (int i = 0; i < n; i++) {
        while (!stack.empty() && 
               temperatures[i] > temperatures[stack.top()]) {
            int prev = stack.top();
            stack.pop();
            result[prev] = i - prev;
        }
        stack.push(i);
    }
    
    return result;
}

File: data structures and algorithms/cs/08_032.cs
using System.Collections.Generic;

public class TemperatureAnalyzer {
    public static int[] DaysUntilWarmer(int[] temperatures) {
        int n = temperatures.Length;
        int[] result = new int[n];
        var stack = new Stack<int>();

        for (int i = 0; i < n; i++) {
            while (stack.Count > 0 && temperatures[i] > temperatures[stack.Peek()]) {
                int prev = stack.Pop();
                result[prev] = i - prev;
            }
            stack.Push(i);
        }

        return result;
    }
}